### PR TITLE
Sort cloud provider and add alibaba

### PR DESCRIPTION
**Pull request by zhangxiaoyu-zidif**
>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> Remember to delete this note before submitting your pull request.
>
> For pull requests on 1.17 Features: set Milestone to 1.17 and Base Branch to dev-1.17
>
> For pull requests on Chinese localization, set Base Branch to release-1.16
> Feel free to ask questions in #kubernetes-docs-zh
>
> For pull requests on Korean Localization: set Base Branch to dev-1.16-ko.\<latest team milestone>
>
> If you need Help on editing and submitting pull requests, visit:
> https://kubernetes.io/docs/contribute/start/#improve-existing-content.
>
> If you need Help on choosing which branch to use, visit:
> https://kubernetes.io/docs/contribute/start#choose-which-git-branch-to-use.
>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>

### CODE REVIEW

**Comment by zhangxiaoyu-zidif** [comment]
fix #18409 #18410

File: content/en/docs/concepts/cluster-administration/cloud-providers.md [hunk 1]
```diff
@@ -52,6 +52,12 @@ for the [kube-apiserver](/docs/admin/kube-apiserver/), [kube-controller-manager]
 For all external cloud providers, please follow the instructions on the individual repositories,
 which are listed under their headings below, or one may view [the list of all repositories](https://github.com/kubernetes?q=cloud-provider-&type=&language=)
 
+## Alibaba Cloud
+ If you wish to use the external cloud provider, its repository is kubernetes/cloud-provider-alibaba-cloud(link:https://github.com/kubernetes/cloud-provider-alibaba-cloud)
```
Comment (tengqm) [inline content/en/docs/concepts/cluster-administration/cloud-providers.md]: This link format is wrong.

File: content/en/docs/concepts/cluster-administration/cloud-providers.md [hunk 2]
```diff
@@ -52,6 +52,12 @@ for the [kube-apiserver](/docs/admin/kube-apiserver/), [kube-controller-manager]
...
+
+### Node Name
+The AlibabaCloudProvider use ${region-id}.${instance-id} as the name of the Kubernetes Node object which is determined by the kubelet overridden with `--hostname-override=${region-id}.${instance-id} --provider-id=${region-id}.${instance-id}`
```
Comment (tengqm) [inline content/en/docs/concepts/cluster-administration/cloud-providers.md]: This sentence doesn't make sense to me. Can kubelet determine it or not? Is the override a must?

File: content/en/docs/concepts/cluster-administration/cloud-providers.md (hunk 1 above)
Comment (zhangxiaoyu-zidif) [reply content/en/docs/concepts/cluster-administration/cloud-providers.md]: done.

File: content/en/docs/concepts/cluster-administration/cloud-providers.md (hunk 2 above)
Comment (zhangxiaoyu-zidif) [reply content/en/docs/concepts/cluster-administration/cloud-providers.md]: After I checked it with my colleague, `provider-id` is a must, and it's better to add `--hostname-override`. ：） so keep it, that's fine.

**Comment by zhangxiaoyu-zidif** [comment]
ping @andrewsykim  for approve

**Comment by zhangxiaoyu-zidif** [comment]
ping  @zacharysarah  @tengqm  for approve and lgtm

**Comment by zacharysarah** [comment]
@zhangxiaoyu-zidif I'm willing to consider this PR with the understanding that how docs handle third-party content is a [discussion in progress](https://github.com/kubernetes/enhancements/pull/1327) leaning towards removal of third-party content that isn't necessary for Kubernetes to function "in tree". Content about cloud providers, on this page and others, will likely be considered out of tree soon and given a deprecation window.

To be specific, please proceed as if **this content will be removed within six months.** 

I want to avoid a scenario where we approve this content addition now only to later hear complaints later when it's removed or refactored.

Are you willing to abide with this understanding?

**Comment by zhangxiaoyu-zidif** [comment]
Hi @zacharysarah 
The config information about cloud provider should have nothing with third-party content which the PR specified. Cloud provider is a tool for users to deploy Kubernetes on cloud environment instead of bare metal. Those informations should be remained for a rather long time, at least more than 6 months 😄.

**Comment by andrewsykim** [comment]
^ oops ignore me, looks like there in separate commits already

File: content/en/docs/concepts/cluster-administration/cloud-providers.md [hunk 3]
```diff
@@ -52,6 +52,12 @@ for the [kube-apiserver](/docs/admin/kube-apiserver/), [kube-controller-manager]
 For all external cloud providers, please follow the instructions on the individual repositories,
 which are listed under their headings below, or one may view [the list of all repositories](https://github.com/kubernetes?q=cloud-provider-&type=&language=)
 
+## Alibaba Cloud
+ If you wish to use the external cloud provider, its repository is [kubernetes/cloud-provider-alibaba-cloud](https://github.com/kubernetes/cloud-provider-alibaba-cloud)
```
Comment (andrewsykim) [inline content/en/docs/concepts/cluster-administration/cloud-providers.md]: Do you have a guide or tutorial on how to set up the external provider? If so I would suggest linking it here as well.

File: content/en/docs/concepts/cluster-administration/cloud-providers.md (hunk 2 above)
Comment (andrewsykim) [reply content/en/docs/concepts/cluster-administration/cloud-providers.md]: I agree with @tengqm that the wording here is confusing. We should explicitly call it out as a pre-requisite if that's the case.

**Comment by sftim** [comment]
@zhangxiaoyu-zidif would you be willing to revise this based on the feedback so far?

File: content/en/docs/concepts/cluster-administration/cloud-providers.md [hunk 4]
```diff
@@ -52,6 +52,12 @@ for the [kube-apiserver](/docs/admin/kube-apiserver/), [kube-controller-manager]
...
+
+### Node Name
+The AlibabaCloudProvider use ${region-id}.${instance-id} as the name of the Kubernetes Node object which is determined by the kubelet overridden with `--hostname-override=${region-id}.${instance-id} --provider-id=${region-id}.${instance-id}`
```
Comment (Colstuwjx) [inline content/en/docs/concepts/cluster-administration/cloud-providers.md]: BTW, quote the name with `${region-id}.${instance-id}` should be better. Like the following providers, maybe we could change this sentence to:

```
The AlibabaCloudProvider ... . Note that the Kubernetes Node name must ...
```

File: content/en/docs/concepts/cluster-administration/cloud-providers.md (hunk 2 above)
Comment (Colstuwjx) [reply content/en/docs/concepts/cluster-administration/cloud-providers.md]: BTW, quote the name with `${region-id}.${instance-id}` should be better. Like the following providers, maybe we could change this sentence to:

```
The AlibabaCloudProvider ... . Note that the Kubernetes Node name must ...
```

Comment (sftim) [reply content/en/docs/concepts/cluster-administration/cloud-providers.md]: @zhangxiaoyu-zidif - if you can paraphrase or explain in detail what meaning you want here, I think that we can suggest revised wording.

**Comment by zacharysarah** [comment]
@zhangxiaoyu-zidif 

> The config information about cloud provider should have nothing with third-party content which the PR specified. Cloud provider is a tool for users to deploy Kubernetes on cloud environment instead of bare metal.

This is indeed an example of content for a third party--in this case, a cloud provider. As such, it will likely be removed when https://github.com/kubernetes/enhancements/pull/1327 is approved. 

Please `/reopen` when you're willing to confirm your agreement and respond to feedback per @sftim. 

/close

**Pull request closed without merging**